app/models/user.rb

```diff
@@ -55,26 +50,17 @@ def self.authenticate(email, password)
     return auth
   end
 
-  def assign_user_id
-    unless @skip_user_id_assign.present? || self.user_id.present?
-      user = User.order("user_id").last
-      uid = if user && user.user_id && !(User.exists?(user_id: "#{user.user_id.to_i + 1}"))
-              user.user_id.to_i + 1
-            else
-              1
-            end
-      self.user_id = uid.to_s if uid
-    end
-  end
-
   def hash_password
-    if password.present? && password_changed?
-      self.password = Digest::MD5.hexdigest(password)
+    if will_save_change_to_password?
+      self.password = Digest::MD5.hexdigest(self.password)
     end
   end
 
   def generate_token(column)
-    self[column] = Encryption.encrypt_sensitive_value(self.user_id)
-    generate_token(column) if User.exists?(column => self[column])
+    begin
+      self[column] = Encryption.encrypt_sensitive_value(self.id)
+    end while User.exists?(column => self[column])
```

Comment:
Rubocop warning

```
app/models/user.rb:62:9: W: Lint/Loop: Use Kernel#loop with break rather than begin/end/until(or while).
    end while User.exists?(column => self[column])
```